# Mercedes-Benz Greener Manufacturing

## Problem Statement Scenario:


Since the first automobile, the Benz Patent Motor Car in 1886, Mercedes-Benz has stood for important automotive innovations. These include the passenger safety cell with a crumple zone, the airbag, and intelligent assistance systems. Mercedes-Benz applies for nearly 2000 patents per year, making the brand the European leader among premium carmakers. Mercedes-Benz is the leader in the premium car industry. With a huge selection of features and options, customers can choose the customized Mercedes-Benz of their dreams.

To ensure the safety and reliability of every unique car configuration before they hit the road, the company’s engineers have developed a robust testing system. As one of the world’s biggest manufacturers of premium cars, safety and efficiency are paramount on Mercedes-Benz’s production lines. However, optimizing the speed of their testing system for many possible feature combinations is complex and time-consuming without a powerful algorithmic approach.

You are required to reduce the time that cars spend on the test bench. Others will work with a dataset representing different permutations of features in a Mercedes-Benz car to predict the time it takes to pass testing. Optimal algorithms will contribute to faster testing, resulting in lower carbon dioxide emissions without reducing Mercedes-Benz’s standards.

### Following actions should be performed:

- If for any column(s), the variance is equal to zero, then you need to remove those variable(s).
- Check for null and unique values for test and train sets.
- Apply label encoder.
- Perform dimensionality reduction.
- Predict your test_df values using XGBoost.

In [1]:
# importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import warnings
from warnings import filterwarnings
filterwarnings('ignore')
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [2]:
# importing data set
train = pd.read_csv("train.csv")

In [3]:
train.head()

ID       y  X0 X1  X2 X3 X4 X5 X6 X8  X10  X11  X12  X13  X14  X15  X16  \
0   0  130.81   k  v  at  a  d  u  j  o    0    0    0    1    0    0    0   
1   6   88.53   k  t  av  e  d  y  l  o    0    0    0    0    0    0    0   
2   7   76.26  az  w   n  c  d  x  j  x    0    0    0    0    0    0    0   
3   9   80.62  az  t   n  f  d  x  l  e    0    0    0    0    0    0    0   
4  13   78.02  az  v   n  f  d  h  d  n    0    0    0    0    0    0    0   

   X17  X18  X19  X20  X21  X22  X23  X24  X26  X27  X28  X29  X30  X31  X32  \
0    0    1    0    0    1    0    0    0    0    0    0    0    0    1    0   
1    0    1    0    0    0    0    0    0    0    1    0    0    0    1    0   
2    1    0    0    0    0    0    0    0    0    1    1    1    0    1    0   
3    0    0    0    0    0    0    0    0    0    1    1    1    0    1    0   
4    0    0    0    0    0    0    0    0    0    1    1    1    0    1    0   

   X33  X34  X35  X36  X37  X38  X39  X40  X41  X42  X43  X44  X45  X46  X47  \
0    0    0    1    0    1    0    0    0    0    0    0    0    0    1    0   
1    0    0    1    0    1    0    0    0    0    0    0    0    0    0    0   
2    0    0    1    0    1    0    0    0    0    0    1    0    0    1    0   
3    0    0    1    0    1    0    0    0    0    0    1    0    0    1    0   
4    0    0    1    0    1    0    0    0    0    0    1    0    0    1    0   

   X48  X49  X50  X51  X52  X53  X54  X55  X56  X57  X58  X59  X60  X61  X62  \
0    0    0    0    0    0    0    0    0    0    0    1    0    0    0    0   
1    0    0    0    1    0    0    0    0    0    0    0    0    0    1    0   
2    0    0    0    1    0    0    1    0    0    0    1    0    0    1    0   
3    0    0    0    0    0    0    1    0    0    0    0    0    0    1    0   
4    0    0    0    1    0    0    1    0    0    0    0    0    0    1    0   

   X63  X64  X65  X66  X67  X68  X69  X70  X71  X73  X74  X75  X76  X77  X78  \
0    0    0    0    0    0    1    0    1    0    0    1    0    0    0    0   
1    0    0    0    0    0    0    0    1    0    0    1    0    0    0    0   
2    0    0    0    0    0    1    0    1    1    0    1    1    1    0    0   
3    0    0    0    0    0    0    0    1    1    0    1    0    1    0    0   
4    0    0    0    0    0    0    0    1    0    0    1    0    1    0    0   

   X79  X80  X81  X82  X83  X84  X85  X86  X87  X88  X89  X90  X91  X92  X93  \
0    0    0    0    0    0    0    1    0    0    0    0    0    0    0    0   
1    0    1    0    0    0    0    1    0    0    0    0    0    0    0    0   
2    0    1    0    0    0    1    1    0    0    0    0    0    0    0    0   
3    0    1    0    0    0    1    0    0    0    0    0    0    0    0    0   
4    0    1    0    0    0    0    0    0    0    0    0    0    0    0    0   

   X94  X95  X96  X97  X98  X99  X100  X101  X102  X103  X104  X105  X106  \
0    0    0    0    0    0    0     0     0     0     0     0     0     0   
1    0    0    1    0    1    0     1     1     0     0     0     0     0   
2    0    0    1    0    1    0     0     1     0     0     0     0     0   
3    0    0    1    0    1    0     0     1     0     0     0     0     0   
4    0    0    1    0    1    0     0     1     0     0     0     0     0   

   X107  X108  X109  X110  X111  X112  X113  X114  X115  X116  X117  X118  \
0     0     0     0     0     1     0     0     1     0     1     0     1   
1     0     0     0     0     1     0     0     0     0     0     0     1   
2     0     0     0     0     1     0     0     0     0     0     0     0   
3     0     1     0     0     1     0     0     1     0     0     0     0   
4     0     1     0     0     1     0     0     1     0     0     0     0   

   X119  X120  X122  X123  X124  X125  X126  X127  X128  X129  X130  X131  \
0     1     1     0     0     0     0     0     0     1     0     0     1   
1     1     1     0     0     0     0     0     1

#### If for any column(s), the variance is equal to zero, then you need to remove those variable(s).

In [4]:
pd.options.display.float_format = '{:,.4f}'.format
variance = train.var()
variance = variance.reset_index()
variance.columns = ["id","values"]
variance= variance.sort_values("values",ascending=1)
variance

id         values
275  X289         0.0000
315  X330         0.0000
254  X268         0.0000
332  X347         0.0000
97   X107         0.0000
221  X235         0.0000
219  X233         0.0000
276  X290         0.0000
3     X11         0.0000
283  X297         0.0000
279  X293         0.0000
83    X93         0.0000
243  X257         0.0002
193  X207         0.0002
266  X280         0.0002
274  X288         0.0002
24    X33         0.0002
30    X39         0.0002
177  X190         0.0002
256  X270         0.0002
281  X295         0.0002
282  X296         0.0002
85    X95         0.0002
246  X260         0.0002
196  X210         0.0002
245  X259         0.0002
324  X339         0.0002
33    X42         0.0002
190  X204         0.0002
191  X205         0.0002
357  X372         0.0005
368  X384         0.0005
222  X236         0.0005
354  X369         0.0005
7     X15         0.0005
255  X269         0.0005
304  X319         0.0005
113  X124         0.0005
264  X278         0.0005
31    X40         0.0007
317  X332         0.0007
231  X245         0.0007
79    X89         0.0007
303  X318         0.0007
64    X74         0.0007
141  X153         0.0007
50    X59         0.0007
238  X252         0.0007
100  X110         0.0009
77    X87         0.0009
155  X167         0.0009
82    X92         0.0009
148  X160         0.0012
342  X357         0.0012
351  X366         0.0012
73    X83         0.0012
252  X266         0.0014
248  X262         0.0014
172  X184         0.0014
76    X86         0.0014
134  X145         0.0014
263  X277         0.0014
239  X253         0.0014
234  X248         0.0014
51    X60         0.0014
369  X385         0.0014
367  X383         0.0017
81    X91         0.0017
94   X104         0.0019
16    X24         0.0019
58    X67         0.0019
199  X213         0.0019
257  X271         0.0021
338  X353         0.0021
56    X65         0.0021
292  X307         0.0021
95   X105         0.0024
179  X192         0.0024
244  X258         0.0024
267  X281         0.0026
295  X310         0.0026
112  X123         0.0026
13    X21         0.0026
8     X16         0.0026
102  X112         0.0028
185  X199         0.0028
350  X365         0.0028
349  X364         0.0028
226  X240         0.0028
213  X227         0.0031
114  X125         0.0031
320  X335         0.0036
268  X282         0.0040
171  X183         0.0040
297  X312         0.0043
87    X97         0.0043
27    X36         0.0045
21    X30         0.0045
153  X165         0.0045
284  X298         0.0045
285  X299         0.0045
17    X26         0.0050
240  X254         0.0052
216  X230         0.0052
46    X55         0.0052
25    X34         0.0054
198  X212         0.0054
68    X78         0.0057
310  X325         0.0057
202  X216         0.0059
53    X62         0.0059
160  X172         0.0059
223  X237         0.0066
157  X169         0.0066
186  X200         0.0066
355  X370         0.0066
225  X239         0.0068
200  X214         0.0068
44    X53         0.0068
92   X102         0.0068
323  X338         0.0068
229  X243         0.0071
78    X88         0.0071
305  X320         0.0071
111  X122         0.0071
294  X309         0.0071
228  X242         0.0073
80    X90         0.0073
84    X94         0.0073
203  X217         0.0073
366  X382         0.0075
9     X17         0.0075
302  X317         0.0075
235  X249         0.0075
10    X18         0.0078
365  X380         0.0080
207  X221         0.0080
326  X341         0.0080
329  X344         0.0085
89    X99         0.0085
278  X292         0.0089
253  X267         0.0089
308  X323         0.0092
364  X379         0.0094
293  X308         0.0094
161  X173         0.0096
260  X274         0.0099
182  X196         0.0101
277  X291         0.0103
23    X32         0.0110
35    X44         0.0113
288  X302         0.0113
32    X41         0.0113
54    X63         0.0113
181  X195         0.0115
67    X77         0.0124
38    X47         0.0127
96   X106         0.0129
290  X305         0.0131
2     X

In [5]:
# Our aim is to eliminate the variables that have a variance of 0 or a variance that is considered low, specifically less than 0.2.
var = variance.loc[variance["values"] < 0.2,"id"]
train_new = train.drop(var,axis=1)
train_new.drop("ID",axis=1,inplace=True)
train_new.head()

y  X0 X1  X2 X3 X4 X5 X6 X8  X14  X27  X46  X51  X58  X64  X85  X100  \
0 130.8100   k  v  at  a  d  u  j  o    0    0    1    0    1    0    1     0   
1  88.5300   k  t  av  e  d  y  l  o    0    1    0    1    0    0    1     1   
2  76.2600  az  w   n  c  d  x  j  x    0    1    1    1    1    0    1     0   
3  80.6200  az  t   n  f  d  x  l  e    0    1    1    0    0    0    0     0   
4  78.0200  az  v   n  f  d  h  d  n    0    1    1    1    0    0    0     0   

   X115  X118  X119  X127  X132  X137  X156  X157  X163  X171  X178  X186  \
0     0     1     1     0     0     1     1     0     0     0     0     0   
1     0     1     1     1     1     0     1     0     0     0     1     0   
2     0     0     0     0     1     1     0     1     1     0     0     0   
3     0     0     0     0     1     0     0     1     0     0     0     0   
4     0     0     0     0     1     0     0     1     0     0     0     0   

   X187  X191  X194  X218  X220  X223  X224  X246  X250  X251  X261  X273  \
0     1     0     1     0     1     0     0     0     0     0     0     1   
1     1     0     1     0     0     0     0     0     1     0     0     1   
2     0     0     1     1     1     1     1     1     1     0     0     1   
3     0     0     1     1     1     1     0     1     1     0     0     1   
4     0     0     1     1     1     1     0     1     1     0     0     1   

   X311  X313  X314  X324  X329  X334  X337  X350  X351  X355  X358  X362  \
0     0     0     0     1     1     1     0     0     0     0     0     0   
1     1     0     0     0     1     0     1     0     0     0     0     0   
2     0     0     0     1     0     1     0     1     0     0     1     0   
3     0     0     0     0     0     0     0     1     0     0     1     0   
4     0     0     0     0     0     1     0     1     0     0     1     0   

   X375  X377  
0     0     1  
1     1     0  
2     0     0  
3     0     0  
4     0     0

### Check for null and unique values for test and train sets.

In [6]:
# Missing values
train_new[train_new.isnull().any(axis=1)]

Empty DataFrame
Columns: [y, X0, X1, X2, X3, X4, X5, X6, X8, X14, X27, X46, X51, X58, X64, X85, X100, X115, X118, X119, X127, X132, X137, X156, X157, X163, X171, X178, X186, X187, X191, X194, X218, X220, X223, X224, X246, X250, X251, X261, X273, X311, X313, X314, X324, X329, X334, X337, X350, X351, X355, X358, X362, X375, X377]
Index: []

#We don't have any missing values in the data

In [7]:
c = train_new.corr().abs()
s = c.unstack()
s = pd.DataFrame(s)
s.reset_index(inplace = True)
s.head()

level_0 level_1      0
0       y       y 1.0000
1       y     X14 0.1936
2       y     X27 0.0535
3       y     X46 0.1360
4       y     X51 0.2300

In [8]:
s["flag"] = np.where(s["level_0"] == s["level_1"],"same","not same")
s.columns.values[2] = "corr"
s.head()

level_0 level_1   corr      flag
0       y       y 1.0000      same
1       y     X14 0.1936  not same
2       y     X27 0.0535  not same
3       y     X46 0.1360  not same
4       y     X51 0.2300  not same

In [9]:
# Remove the variables with correlation more than .9
name = s.loc[(s["corr"] > .9) & (s["flag"] != "same") ,"level_1"]
final_name = name.unique()

In [10]:
train_new_updated = train_new.drop(final_name,axis=1)
train_new_updated.head()

y  X0 X1  X2 X3 X4 X5 X6 X8  X27  X46  X51  X64  X85  X100  X115  \
0 130.8100   k  v  at  a  d  u  j  o    0    1    0    0    1     0     0   
1  88.5300   k  t  av  e  d  y  l  o    1    0    1    0    1     1     0   
2  76.2600  az  w   n  c  d  x  j  x    1    1    1    0    1     0     0   
3  80.6200  az  t   n  f  d  x  l  e    1    1    0    0    0     0     0   
4  78.0200  az  v   n  f  d  h  d  n    1    1    1    0    0     0     0   

   X127  X132  X163  X171  X191  X218  X220  X223  X224  X273  X313  X329  \
0     0     0     0     0     0     0     1     0     0     1     0     1   
1     1     1     0     0     0     0     0     0     0     1     0     1   
2     0     1     1     0     0     1     1     1     1     1     0     0   
3     0     1     0     0     0     1     1     1     0     1     0     0   
4     0     1     0     0     0     1     1     1     0     1     0     0   

   X350  X351  X355  X375  X377  
0     0     0     0     0     1  
1     0     0     0     1     0  
2     1     0     0     0     0  
3     1     0     0     0     0  
4     1     0     0     0     0

In [11]:
train_new_updated.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4209 entries, 0 to 4208
Data columns (total 33 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   y       4209 non-null   float64
 1   X0      4209 non-null   object 
 2   X1      4209 non-null   object 
 3   X2      4209 non-null   object 
 4   X3      4209 non-null   object 
 5   X4      4209 non-null   object 
 6   X5      4209 non-null   object 
 7   X6      4209 non-null   object 
 8   X8      4209 non-null   object 
 9   X27     4209 non-null   int64  
 10  X46     4209 non-null   int64  
 11  X51     4209 non-null   int64  
 12  X64     4209 non-null   int64  
 13  X85     4209 non-null   int64  
 14  X100    4209 non-null   int64  
 15  X115    4209 non-null   int64  
 16  X127    4209 non-null   int64  
 17  X132    4209 non-null   int64  
 18  X163    4209 non-null   int64  
 19  X171    4209 non-null   int64  
 20  X191    4209 non-null   int64  
 21  X218    4209 non-null   int64  
 22  

## Create dummy variables

In [12]:
char = train_new_updated.select_dtypes(exclude='number')
num = train_new_updated.select_dtypes(include='number')

In [13]:
char1 = pd.get_dummies(char.astype(str),drop_first=True)
char1.head()

X0_aa  X0_ab  X0_ac  X0_ad  X0_af  X0_ai  X0_aj  X0_ak  X0_al  X0_am  \
0      0      0      0      0      0      0      0      0      0      0   
1      0      0      0      0      0      0      0      0      0      0   
2      0      0      0      0      0      0      0      0      0      0   
3      0      0      0      0      0      0      0      0      0      0   
4      0      0      0      0      0      0      0      0      0      0   

   X0_ao  X0_ap  X0_aq  X0_as  X0_at  X0_au  X0_aw  X0_ax  X0_ay  X0_az  X0_b  \
0      0      0      0      0      0      0      0      0      0      0     0   
1      0      0      0      0      0      0      0      0      0      0     0   
2      0      0      0      0      0      0      0      0      0      1     0   
3      0      0      0      0      0      0      0      0      0      1     0   
4      0      0      0      0      0      0      0      0      0      1     0   

   X0_ba  X0_bc  X0_c  X0_d  X0_e  X0_f  X0_g  X0_h  X0_i  X0_j  X0_k  X0_l  \
0      0      0     0     0     0     0     0     0     0     0     1     0   
1      0      0     0     0     0     0     0     0     0     0     1     0   
2      0      0     0     0     0     0     0     0     0     0     0     0   
3      0      0     0     0     0     0     0     0     0     0     0     0   
4      0      0     0     0     0     0     0     0     0     0     0     0   

   X0_m  X0_n  X0_o  X0_q  X0_r  X0_s  X0_t  X0_u  X0_v  X0_w  X0_x  X0_y  \
0     0     0     0     0     0     0     0     0     0     0     0     0   
1     0     0     0     0     0     0     0     0     0     0     0     0   
2     0     0     0     0     0     0     0     0     0     0     0     0   
3     0     0     0     0     0     0     0     0     0     0     0     0   
4     0     0     0     0     0     0     0     0     0     0     0     0   

   X0_z  X1_aa  X1_ab  X1_b  X1_c  X1_d  X1_e  X1_f  X1_g  X1_h  X1_i  X1_j  \
0     0      0      0     0     0     0     0     0     0     0     0     0   
1     0      0      0     0     0     0     0     0     0     0     0     0   
2     0      0      0     0     0     0     0     0     0     0     0     0   
3     0      0      0     0     0     0     0     0     0     0     0     0   
4     0      0      0     0     0     0     0     0     0     0     0     0   

   X1_k  X1_l  X1_m  X1_n  X1_o  X1_p  X1_q  X1_r  X1_s  X1_t  X1_u  X1_v  \
0     0     0     0     0     0     0     0     0     0     0     0     1   
1     0     0     0     0     0     0     0     0     0     1     0     0   
2     0     0     0     0     0     0     0     0     0     0     0     0   
3     0     0     0     0     0     0     0     0     0     1     0     0   
4     0     0     0     0     0     0     0     0     0     0     0     1   

   X1_w  X1_y  X1_z  X2_aa  X2_ac  X2_ae  X2_af  X2_ag  X2_ah  X2_ai  X2_ak  \
0     0     0     0      0      0      0      0      0      0      0      0   
1     0     0     0      0      0      0      0      0      0      0      0   
2     1     0     0      0      0      0      0      0      0      0      0   
3     0     0     0      0      0      0      0      0      0      0      0   
4     0     0     0      0      0      0      0      0      0      0      0   

   X2_al  X2_am  X2_an  X2_ao  X2_ap  X2_aq  X2_ar  X2_as  X2_at  X2_au  \
0      0      0      0      0      0      0      0      0      1      0   
1      0      0      0      0      0      0      0      0      0      0   
2      0      0      0      0      0      0      0      0      0      0   
3      0      0      0      0      0      0      0      0      0      0   
4      0      0      0      0      0      0      0      0      0      0   

   X2_av  X2_aw  X2_ay  X2_b  X2_c  X2_d  X2_e  X2_f  X2_g  X2_h  X2_i  X2_j  \
0      0      0      0     0     0     0     0     0     0     0     0     0   
1      1      0      0     0     0     0     0     0     0     0     0     0   
2      0      0      0 

In [14]:
data3 = pd.concat([char1,num],axis=1)
data3.head()

X0_aa  X0_ab  X0_ac  X0_ad  X0_af  X0_ai  X0_aj  X0_ak  X0_al  X0_am  \
0      0      0      0      0      0      0      0      0      0      0   
1      0      0      0      0      0      0      0      0      0      0   
2      0      0      0      0      0      0      0      0      0      0   
3      0      0      0      0      0      0      0      0      0      0   
4      0      0      0      0      0      0      0      0      0      0   

   X0_ao  X0_ap  X0_aq  X0_as  X0_at  X0_au  X0_aw  X0_ax  X0_ay  X0_az  X0_b  \
0      0      0      0      0      0      0      0      0      0      0     0   
1      0      0      0      0      0      0      0      0      0      0     0   
2      0      0      0      0      0      0      0      0      0      1     0   
3      0      0      0      0      0      0      0      0      0      1     0   
4      0      0      0      0      0      0      0      0      0      1     0   

   X0_ba  X0_bc  X0_c  X0_d  X0_e  X0_f  X0_g  X0_h  X0_i  X0_j  X0_k  X0_l  \
0      0      0     0     0     0     0     0     0     0     0     1     0   
1      0      0     0     0     0     0     0     0     0     0     1     0   
2      0      0     0     0     0     0     0     0     0     0     0     0   
3      0      0     0     0     0     0     0     0     0     0     0     0   
4      0      0     0     0     0     0     0     0     0     0     0     0   

   X0_m  X0_n  X0_o  X0_q  X0_r  X0_s  X0_t  X0_u  X0_v  X0_w  X0_x  X0_y  \
0     0     0     0     0     0     0     0     0     0     0     0     0   
1     0     0     0     0     0     0     0     0     0     0     0     0   
2     0     0     0     0     0     0     0     0     0     0     0     0   
3     0     0     0     0     0     0     0     0     0     0     0     0   
4     0     0     0     0     0     0     0     0     0     0     0     0   

   X0_z  X1_aa  X1_ab  X1_b  X1_c  X1_d  X1_e  X1_f  X1_g  X1_h  X1_i  X1_j  \
0     0      0      0     0     0     0     0     0     0     0     0     0   
1     0      0      0     0     0     0     0     0     0     0     0     0   
2     0      0      0     0     0     0     0     0     0     0     0     0   
3     0      0      0     0     0     0     0     0     0     0     0     0   
4     0      0      0     0     0     0     0     0     0     0     0     0   

   X1_k  X1_l  X1_m  X1_n  X1_o  X1_p  X1_q  X1_r  X1_s  X1_t  X1_u  X1_v  \
0     0     0     0     0     0     0     0     0     0     0     0     1   
1     0     0     0     0     0     0     0     0     0     1     0     0   
2     0     0     0     0     0     0     0     0     0     0     0     0   
3     0     0     0     0     0     0     0     0     0     1     0     0   
4     0     0     0     0     0     0     0     0     0     0     0     1   

   X1_w  X1_y  X1_z  X2_aa  X2_ac  X2_ae  X2_af  X2_ag  X2_ah  X2_ai  X2_ak  \
0     0     0     0      0      0      0      0      0      0      0      0   
1     0     0     0      0      0      0      0      0      0      0      0   
2     1     0     0      0      0      0      0      0      0      0      0   
3     0     0     0      0      0      0      0      0      0      0      0   
4     0     0     0      0      0      0      0      0      0      0      0   

   X2_al  X2_am  X2_an  X2_ao  X2_ap  X2_aq  X2_ar  X2_as  X2_at  X2_au  \
0      0      0      0      0      0      0      0      0      1      0   
1      0      0      0      0      0      0      0      0      0      0   
2      0      0      0      0      0      0      0      0      0      0   
3      0      0      0      0      0      0      0      0      0      0   
4      0      0      0      0      0      0      0      0      0      0   

   X2_av  X2_aw  X2_ay  X2_b  X2_c  X2_d  X2_e  X2_f  X2_g  X2_h  X2_i  X2_j  \
0      0      0      0     0     0     0     0     0     0     0     0     0   
1      1      0      0     0     0     0     0     0     0     0     0     0   
2      0      0      0 

## Run the XGBoost model and calculate the mape

In [15]:
from sklearn.model_selection import train_test_split
import xgboost as xg

In [16]:
#segregating ind and dep var
X = data3.drop("y",axis=1)
y = data3.loc[:,"y"]

In [17]:
# Splitting
X_train, X_test, y_train, y_test = train_test_split(X, y,test_size = 0.3, random_state = 185)

In [18]:
# Instantiation
xgb_r = xg.XGBRegressor(objective ='reg:squarederror',n_estimators = 10, seed = 123)

In [19]:
# Fitting the model
xgb_r.fit(X_train, y_train)

XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, gpu_id=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=None, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             n_estimators=10, n_jobs=None, num_parallel_tree=None,
             predictor=None, random_state=None, ...)

In [20]:
# Predict the model
y_pred = xgb_r.predict(X_test)

In [21]:
###### mape
final_predictions = pd.DataFrame()
final_predictions["y_test"] = y_test
final_predictions["y_pred"] = y_pred
final_predictions["mp"] = abs((final_predictions["y_test"]- final_predictions["y_pred"])/final_predictions["y_test"])
final_predictions.to_csv('predictions.csv', index=False)
(final_predictions.mp.mean())*100#mape

4.902028696491781

In [22]:
final_predictions.head()

y_test   y_pred     mp
2728 106.4400  90.2419 0.1522
4083  89.6000  89.5833 0.0002
2813 104.2800 109.0980 0.0462
2795  90.8300  92.0411 0.0133
231   91.1600  92.1301 0.0106

#### We are getting a mape of 4.902 meaning an accuracy of more than 95%